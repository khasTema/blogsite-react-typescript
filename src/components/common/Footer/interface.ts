export interface IfooterConfig {
    footerText: string
}