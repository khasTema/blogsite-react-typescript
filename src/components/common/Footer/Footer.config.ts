import { IfooterConfig } from "./interface"

export const footerConfig: IfooterConfig = {
    footerText: 'This is footer. Yeah!'
}