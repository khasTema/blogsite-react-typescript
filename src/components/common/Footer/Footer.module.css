.footer {
    background-color: var(--secondary-color);
    width: 100%;
    text-align: center;
    padding: 2rem 0;
    color: var(--logo-text-color);
    font-size: .8rem;
    font-weight: var(--fontweight-reg);
}