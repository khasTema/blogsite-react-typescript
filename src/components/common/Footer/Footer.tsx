import React from 'react'
import { footerConfig } from './Footer.config'
import styles from './Footer.module.css'


export const Footer: React.FC = ():JSX.Element => {
  return (
    <footer className={styles.footer}>
      {footerConfig.footerText}
    </footer>
  )
}

