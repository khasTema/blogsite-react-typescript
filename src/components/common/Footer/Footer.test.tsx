import renderer from 'react-test-renderer';
import React from 'react'
import { Footer } from './Footer';
import { JSX } from 'react/jsx-runtime';

jest.mock('./Footer.config');
jest.mock('./Footer.module.css');

const renderTree = (tree: JSX.Element) => renderer.create(tree);
describe('<Footer>', () => {
  it('should render component Footer', () => {
    expect(renderTree(<Footer 
    />).toJSON()).toMatchSnapshot();
  });
  
});