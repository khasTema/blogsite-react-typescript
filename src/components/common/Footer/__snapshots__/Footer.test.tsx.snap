// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> should render component Footer 1`] = `
<footer>
  This is footer. Yeah!
</footer>
`;
